src/System.Runtime/tests/System/ArrayTests.cs

```diff
@@ -976,6 +1088,35 @@ public static void FindLastIndex_Invalid()
             Assert.Throws<ArgumentOutOfRangeException>("startIndex", () => Array.FindLastIndex(new int[3], 3, 1, i => i == 43));
         }
 
+        [Theory]
+        public static void ConvertAll()
+        {
+            var result = Array.ConvertAll<int, int>(new int[] { }, new Converter<int, int>(i => { throw new InvalidOperationException(); }));
+            // Does not throw - no entries
+            Assert.Equal(new int[] { }, result);
+
+            var result2 = Array.ConvertAll<int, string>(new int[] { 1 }, new Converter<int, string>(i => i++.ToString()));
+            Assert.Equal(new string[] { "2" }, result2);
+
+            result2 = Array.ConvertAll<int, string>(new int[] { 1, 2 }, new Converter<int, string>(i => i++.ToString()));
+            Assert.Equal(new string[] { "2", "3" }, result2);
+
+            result2 = Array.ConvertAll<int, string>(new int[] { 1 }, new Converter<int, string>(i => null));
+            Assert.Equal(new string[] { null }, result2);
+
+            // ConvertAll only handles one dimensional arrays
```

Comment:
Same comments as earlier about testing this, adding string args to Assert.Throws with arg exceptions, etc.